### PR TITLE
Feature/dm 343 new maxquant model

### PR DESCRIPTION
**Pull request by wow-such-code**
Updates the parser for MaxQuant data to handle the new model: experimentalDesignTemplate, summary and an unnecessary folder are removed. The sample ids file now needs a valid QBiC project code.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -127,7 +127,7 @@
     <dependency>
       <artifactId>data-model-lib</artifactId>
       <groupId>life.qbic</groupId>
-      <version>2.18.0</version>
+      <version>2.19.0-SNAPSHOT</version>
```
Comment (jenniferboedker) [inline pom.xml:130]: I think this will be done by the workflow, isnt it?

Comment (wow-such-code) [reply pom.xml:130]: increase the dependency? I don't think so, needed that Snapshot to test the new model

**Review by jenniferboedker — CHANGES_REQUESTED** [review]
Hi, thanks for your work!

I found that within the code the summary folder (see my comment) is contained and also the combined folder could in theory be parsed. Why did you keep them? I think we would need to remove them and make a major release afterwards. What do you think?

File: pom.xml (hunk 1 above)
Comment (jenniferboedker) [reply pom.xml:130]: ahh my bad I did not look closely 🙈

File: src/test/groovy/life/qbic/utils/MaxQuantParserSpec.groovy [hunk 2]
```diff
@@ -32,35 +32,29 @@ class MaxQuantParserSpec extends Specification {
...
         assert maxQuantRunResult.allPeptides.getName()== "allPeptides.txt"
 
-        assert maxQuantRunResult.evidence.getRelativePath() == "./combined/txt/evidence.txt"
+        assert maxQuantRunResult.evidence.getRelativePath() == "./txt/evidence.txt"
         assert maxQuantRunResult.evidence.getName()== "evidence.txt"
 
-        assert maxQuantRunResult.experimentalDesignTemplate.getRelativePath() == "./combined/txt/experimentalDesignTemplate.txt"
-        assert maxQuantRunResult.experimentalDesignTemplate.getName()== "experimentalDesignTemplate.txt"
-
-        assert maxQuantRunResult.parameters.getRelativePath() == "./combined/txt/parameters.txt"
```
[the remark this answers is not in the document]
Comment (wow-such-code) [reply src/test/groovy/life/qbic/utils/MaxQuantParserSpec.groovy:48]: design file is removed, parameters file changes folders

File: src/main/groovy/life/qbic/utils/MaxQuantParser.groovy [hunk 3]
```diff
@@ -137,12 +135,12 @@ class MaxQuantParser implements DatasetParser<MaxQuantRunResult> {
      * Method which adapts the parsed content of the txt directory in place to the expected file structure.
      * @see {<a href="https://github.com/qbicsoftware/data-model-lib/blob/master/src/test/resources/examples/resultset/maxquant/valid-resultset-example.json">valid datastructure example</a>}
      *
-     * After parsing the files of the txt directory are contained in the children property of the combined directory, which itself is contained in the root directory.
+     * After parsing the files of the txt directory and a potential intermediate directory, which itself is contained in the root directory.
```
[the remark this answers is not in the document]
Comment (wow-such-code) [reply src/main/groovy/life/qbic/utils/MaxQuantParser.groovy]: thank you, has been addressed

File: src/main/groovy/life/qbic/utils/MaxQuantParser.groovy [hunk 4]
```diff
@@ -151,6 +149,9 @@ class MaxQuantParser implements DatasetParser<MaxQuantRunResult> {
             if (map.get("name") == "combined") {
                 combinedFolderInformation = map.get("children")
             }
+            if (map.get("name") == "txt") {
+                txtFolderInformation = map.get("children")  as List<Map>
+            }
         }
         if (combinedFolderInformation) {
```
[the remark this answers is not in the document]
Comment (wow-such-code) [reply src/main/groovy/life/qbic/utils/MaxQuantParser.groovy]: removed both of them

**Review by jenniferboedker — APPROVED** [review]
LGTM:)

**Pull request merged**